pkg/databaseclaim/awsprovider.go

```diff
@@ -743,6 +742,30 @@ func (r *DatabaseClaimReconciler) deleteExternalResourcesAWS(ctx context.Context
 	return nil
 }
 
+func (r *DatabaseClaimReconciler) cloudDatabaseExistsAWS(ctx context.Context, dbHostName string) bool {
```

Comment:
Can the method pointer be removed so this code can be unit tested?
```suggestion
func cloudDatabaseExistsAWS(ctx context.Context, cli client.Getter, dbHostName string) bool {
```

Then a mock client can be used to verify functionality in here.

```
func TestCRExists(t *testing.T) {
  ok := cloudDatababaseExistsAWS(ctx, getter, "existingclaim")
  ...
}
```